docs/java-rest/high-level/apis/putmapping.asciidoc

```diff
@@ -0,0 +1,71 @@
+[[java-rest-high-put-mapping]]
+=== Put Mapping API
+
+[[java-rest-high-put-mapping-request]]
+==== Put Mapping Request
+
+A `PutMappingRequest` requires an `index` argument, and a type:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[put-mapping-request]
+--------------------------------------------------
+<1> The index to add the mapping to
+<2> The type to create (or update)
+
+==== Mapping source
+A description of the fields to create on the mapping; if not defined, the mapping will default to empty.
```

Comment:
should we list this under the optional arguments? After all it is optional?

Reply:
Hmm. Technically it's optional; but a PutMapping request without a source just adds a new type; which, in ES 6.x+ world, is basically meaningless.